Training/Training/intlist.c

```diff
@@ -0,0 +1,285 @@
+/// the next node and the previous node is freed.
+/// </summary>
+/// <param name="list"></param>
+/// <returns>Error code (if any)</returns>
+int Delete (struct LinkedList* list) {
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	if (list->head == NULL) {
+		list->isDeleted = true;
+		return 0;
+	}
+
+	struct Node* temp = NULL;
+
+	while (list->head != NULL) {
+		temp = list->head;
+		list->head = (list->head)->next;
+		free (temp);
+	}
+	list->isDeleted = true;
+	return 0;
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Adds nodes to an existing list. 
+/// </summary>
+/// <param name="list"></param>
+/// <param name="data"></param>
+/// <returns>Error code (if any)</returns>
+int Add (struct LinkedList* list, int data) {
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	struct Node* newNode = CreateNode (data);
+
+	// in case this is the first node, make the newNode as
+	// the head of the LinkedList
+	if (list->head == NULL) {
+		list->head = newNode;
+		return 0;
+	}
+	// Create a pointer to iterate till the last node
+	struct Node* current = list->head;
+	while (current->next != NULL) {
+		current = current->next;
+	}
+	// make the next of the tail to the newNode
+	current->next = newNode;
+	return 0;
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Inserts a node at the specified index with the specified value into the list
+/// </summary>
+/// <param name="list"></param>
+/// <param name="ind"></param>
+/// <param name="data"></param>
+/// <returns>Error code (if any)</returns>
+int Insert (struct LinkedList* list, int ind, int data) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	struct Node* newNode = CreateNode (data);
+
+	// If inserting at the beginning
+	if (ind == 0) {
+		newNode->next = list->head;
+		list->head = newNode;
+		return 0;
+	}
+
+	struct Node* current = list->head;
+	for (int i = 0; i < ind - 1 && current != NULL; ++i) {
+		current = current->next;
+	}
+
+	// If the position is out of bounds
+	if (current == NULL || ind < 0) {
+		free (newNode);
+		return E_INVALID_INDEX;
+	}
+
+	newNode->next = current->next;
+	current->next = newNode;
+	return 0;
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Removes the node at a specified index from the list.
+/// </summary>
+/// <param name="list"></param>
+/// <param name="ind"></param>
+/// <returns>Error code (if any)</returns>
+int RemoveAt (struct LinkedList* list, int ind) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	struct Node* temp = list->head;
+	struct Node* prev = NULL;
+
+	// Base case if linked list is empty
+	if (temp == NULL)
+		return E_EMPTY_LIST;
+
+	// Case 1: Head is to be deleted
+	if (ind == 0) {
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+
+	// Case 2: Node to be deleted is in middle

```

Comment:
You're already returning if temp ==  NULL, there is no need to check again. 
You should check if ind is within valid range (in which case you know it will never be null).